src/server/req_manager.c

```diff
@@ -1677,6 +1681,65 @@ mgr_server_unset(struct batch_request *preq)
 		if (server.sv_attr[SVR_ATR_DefaultChunk].at_flags & ATR_VFLAG_MODIFY) {
 			(void)deflt_chunk_action(&server.sv_attr[SVR_ATR_DefaultChunk], (void *)&server, ATR_ACTION_ALTER);
 		}
+		/* Now set the default values on some of the unset attributes */
+		plist = (svrattrl *)GET_NEXT(preq->rq_ind.rq_manager.rq_attr);
+		for (plist = (svrattrl *)GET_NEXT(preq->rq_ind.rq_manager.rq_attr);
+			plist != NULL; plist = (struct svrattrl *)GET_NEXT(plist->al_link)) {
+			if (strcasecmp(plist->al_name, ATTR_logevents) == 0) {
+				char dflt_log_event[64];
+				sprintf(dflt_log_event, "%d", PBSEVENT_ERROR | PBSEVENT_SYSTEM | PBSEVENT_ADMIN | PBSEVENT_JOB |
+								PBSEVENT_JOB_USAGE | PBSEVENT_SECURITY | PBSEVENT_SCHED |
+								PBSEVENT_DEBUG | PBSEVENT_DEBUG2);
+				set_attr_svr(&(server.sv_attr[(int)SRV_ATR_log_events]), &svr_attr_def[(int) SRV_ATR_log_events],
+					     dflt_log_event);
+			}
+			else if (strcasecmp(plist->al_name, ATTR_mailfrom) == 0)
+				set_attr_svr(&(server.sv_attr[(int)SRV_ATR_mailfrom]),
+					    &svr_attr_def[(int)SRV_ATR_mailfrom], "adm");
+			else if (strcasecmp(plist->al_name, ATTR_queryother) == 0)
+				set_attr_svr(&(server.sv_attr[(int)SRV_ATR_query_others]),
+					    &svr_attr_def[(int)SRV_ATR_query_others], "TRUE");
+			else if (strcasecmp(plist->al_name, ATTR_schediteration) == 0)
+				set_attr_svr(&(server.sv_attr[(int)SRV_ATR_scheduler_iteration]),
+					    &svr_attr_def[(int)SRV_ATR_scheduler_iteration], TOSTR(PBS_SCHEDULE_CYCLE));
+			else if(strcasecmp(plist->al_name, ATTR_ResvEnable) == 0)
+				set_attr_svr(&(server.sv_attr[(int)SRV_ATR_ResvEnable]),
+					    &svr_attr_def[(int)SRV_ATR_ResvEnable], "TRUE");
+			else if(strcasecmp(plist->al_name, ATTR_maxarraysize) == 0)
+				set_attr_svr(&(server.sv_attr[(int)SVR_ATR_maxarraysize]),
+					    &svr_attr_def[(int)SVR_ATR_maxarraysize], TOSTR(PBS_MAX_ARRAY_JOB_DFL));
+			else if(strcasecmp(plist->al_name, ATTR_max_concurrent_prov) == 0)
+				set_attr_svr(&(server.sv_attr[(int)SRV_ATR_max_concurrent_prov]),
+					    &svr_attr_def[(int)SRV_ATR_max_concurrent_prov], TOSTR(PBS_MAX_CONCURRENT_PROV));
+			else if(strcasecmp(plist->al_name, ATTR_EligibleTimeEnable) == 0)
+				set_attr_svr(&(server.sv_attr[(int)SRV_ATR_EligibleTimeEnable]),
+					    &svr_attr_def[(int)SRV_ATR_EligibleTimeEnable], "FALSE");
+			else if(strcasecmp(plist->al_name, ATTR_license_linger) == 0)
+				set_attr_svr(&(server.sv_attr[(int)SRV_ATR_license_linger]),
+					    &svr_attr_def[(int)SRV_ATR_license_linger], TOSTR(PBS_LIC_LINGER_TIME));
+			else if ((strcasecmp(plist->al_name, ATTR_DefaultChunk) == 0) ||
+				 (strcasecmp(plist->al_name, ATTR_rescdflt) == 0)){
+				if (plist->al_resc != NULL && strcasecmp(plist->al_resc, "ncpus") == 0) {
+					svrattrl *tm_list;
+					tm_list = attrlist_create(plist->al_name, "ncpus", 1);
+					if (tm_list == NULL) {
+						reply_badattr(-1, bad_attr, plist, preq);
+					}
+					tm_list->al_link.ll_next->ll_struct = NULL;
```

Comment:
Do you need to free anything first?

Reply:
Since this is newly created list one 1 element, I don't think there is anything to free.